packages/rest/test/unit/coercion/utils.ts

```diff
@@ -0,0 +1,95 @@
+import {OperationObject, ParameterObject} from '@loopback/openapi-v3-types';
+
+import {
+  ShotRequestOptions,
+  expect,
+  stubExpressContext,
+} from '@loopback/testlab';
+
+import {
+  PathParameterValues,
+  Request,
+  Route,
+  createResolvedRoute,
+  parseOperationArgs,
+  ResolvedRoute,
+} from '../../..';
+import * as HttpErrors from 'http-errors';
+
+function givenOperationWithParameters(params?: ParameterObject[]) {
+  return <OperationObject>{
+    'x-operation-name': 'testOp',
+    parameters: params,
+    responses: {},
+  };
+}
+
+function givenRequest(options?: ShotRequestOptions): Request {
+  return stubExpressContext(options).request;
+}
+
+function givenResolvedRoute(
+  spec: OperationObject,
+  pathParams: PathParameterValues = {},
+): ResolvedRoute {
+  const route = new Route('get', '/', spec, () => {});
+  return createResolvedRoute(route, pathParams);
+}
+
+export interface TestArgs<T> {
+  paramSpec: ParameterObject;
+  rawValue: string | undefined | object;
+  expectedResult: T;
+  caller: string;
+  expectError: boolean;
+  opts: TestOptions;
+}
+
+export type TestOptions = {
+  testName?: string;
+};
+
+export async function testCoercion<T>(config: TestArgs<T>) {
+  try {
+    const req = givenRequest();
+    const spec = givenOperationWithParameters([config.paramSpec]);
+    const route = givenResolvedRoute(spec, {aparameter: config.rawValue});
+
+    if (config.expectError) {
+      try {
+        await parseOperationArgs(req, route);
+        throw new Error("'parseOperationArgs' should throw error!");
+      } catch (err) {
+        expect(err).to.eql(config.expectedResult);
+      }
```

Comment:
I think this block could be simplified using `rejectedWith`.

```ts
if (config.expectError) {
  expect(parseOperationArgs(req, route))
    .to.be.rejectedWith(config.expectedResult);
}
```

Not a big deal though, if my proposal does not work then I am ok with your current solution too.

Reply:
@bajtos I have also tried a lot of approaches to test the error for an async function, including `rejectedWith`, but none of them work, therefore `try{...}catch(err){...}` is used here.

E.g. by using `rejectedWith`:
```
expect(await parseOperationArgs(req, route)).to.be.rejectedWith(config.expectedResult);
```
It fails the test directly instead of compare the error message. While I would definitely like to try more to simplify the error msg test in the coming PRs, if time allows.

Reply:
It's important to pass the promise returned by `parseOperationArgs` directly to `expect` and don't await it!

Your code which does not work:

```ts
expect(await parseOperationArgs(/*...*/)).to.be.rejectedWith(/*...*/);
```

Correct usage - let `expect()` handle the promise returned by `parseOperationArgs`, await the result of `expect` call:

```ts
await expect(parseOperationArgs(/*...*/)).to.be.rejectedWith(/*...*/);
```

A more verbose alternative:

```ts
expect(async () => { await parseOperationArgs(/*...*/); })
  .to.be.rejectedWith(/*...*/);

// a sync version for better understanding of the principle
expect(() => { doSomethingSync(/*...*/); })
  .to.throw(/*...*/);
```